modules/gdscript/gdscript_editor.cpp

```diff
@@ -1425,6 +1425,12 @@ static bool _guess_identifier_type_from_base(GDScriptCompletionContext &p_contex
 									// Variable used in the same expression
 									return false;
 								}
+
+								if (m.data_type.has_type) {
```

Comment:
`_guess_expression_type` should come before this. The reason is that guessing can get the actual value type while this can only get the user-defined type. So if the set type is `Node` but the actual value set is `Node2D` then guessing will assume a `Node2D` but using the defined type will only complete with `Node` members.

Reply:
It would be likely to remove this part because it was only for the `onready var` assignment of a child node. i.e `onready var test_node: Control = $Some/Control`. Which works while the current script belongs to the currently open scene.